### PR TITLE
Page editor extensions

### PR DESCRIPTION
**Pull request by pawel-boguski-ds**

### CODE REVIEW

File: docs/docs/developers/development/page-editor/extending-editor/index.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
+# Extending Page Editor
+
+## Overview
+
+Developers may need to extend functionality of _Page editor_. It can be done by providing _Editor extension_.
+_Editor extension_ is delivered as JavaScript and use _Page editor_ API.
+
+!!! Info "Note"
+        Detailed specification of _Page editor_ API will be provided together with next versions of WebSight CMS.
```
Comment (kamil-kwarciak) [inline docs/docs/developers/development/page-editor/extending-editor/index.md:9]: @pawel-boguski-ds, do we mean the very next release 1.9.0, or just sooner or later?

File: docs/docs/developers/development/page-editor/extending-editor/index.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+## Registering extension
+
+To define the _Editor extension_ you must create JavaScript file which will be loaded in  the_Page editor_ by the _Web Fragments_.
```
Comment (kamil-kwarciak) [inline docs/docs/developers/development/page-editor/extending-editor/index.md:13]:
```suggestion
To define the _Editor extension_ you must create JavaScript file which will be loaded in  the _Page editor_ by the _Web Fragments_.
```

**Review by michalcukierman — APPROVED** [review]

File: docs/docs/developers/development/page-editor/extending-editor/index.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
+# Extending Page Editor
+
+## Overview
+
+Developers may need to extend functionality of _Page editor_. It can be done by providing _Editor extension_.
```
Comment (kamil-kwarciak) [inline docs/docs/developers/development/page-editor/extending-editor/index.md:5]:
```suggestion
Developers may need to extend the functionality of _Page editor_. You can provide _Editor extension_ to achieve it. 
```

File: docs/docs/developers/development/page-editor/extending-editor/index.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+[dynamic import](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import) in the application runtime.
+The _Web Fragments_ scripts should provide [default export](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/export)
+delivering object, function, etc. depending on the given case - _Web Fragments_ key that given fragment is related too.
+Expected type of default export should be checked in specification of given extension point.
+In number parts of application the _Web Fragments_ with particular keys are imported allowing to extend the WebSight CMS.
+
+The key of _Web Fragments_ loaded by _Page editor_ to get extensions is `websight.editor.spi.extension`.
+Expected default export should provide object with `init` function with single argument which will be _Page editor_ object.
+
+To register the fragment provide the OSGi components implementing `pl.ds.websight.fragments.registry.WebFragment` interface
```
Comment (kamil-kwarciak) [inline docs/docs/developers/development/page-editor/extending-editor/index.md:25]:
```suggestion
To register the fragment, provide the OSGi components implementing `pl.ds.websight.fragments.registry.WebFragment` interface
```

File: docs/docs/developers/development/page-editor/extending-editor/index.md (hunk 1 above)
Comment (pawel-boguski-ds) [reply docs/docs/developers/development/page-editor/extending-editor/index.md:9]: Sooner or later.

**Pull request merged**